### PR TITLE
Adds support for headerLayoutPreset and headerBackTitleEnabled

### LINKED ISSUE #3153: Header title not centered when there is a right/left button
### Version
Tell us which versions you are using: 

- react-native-router-flux@4.0.0-beta.31
- react-native@0.56.0

### Expected behaviour
Header title to be centered

### Actual behaviour
image

### PR DESCRIPTION
**Pull request by daviscabral**
Closes #3153

### CODE REVIEW

**Comment by daviscabral** [comment]
@aksonov if possible, include this one too in the next release.

File: src/navigationStore.js [hunk 1]
```diff
@@ -195,15 +197,17 @@ function createNavigationOptions(params) {
...
-      headerTintColor: navBarButtonColor || props.tintColor || navigationParams.tintColor || navigationParams.headerTintColor,
-      headerTitleStyle: headerTitleStyle || titleStyle,
-      title: getValue(navigationParams.title || title || getTitle, state),
+      headerBackImage: navigationParams.backButtonImage || backButtonImage,
       headerBackTitle: getValue(navigationParams.backTitle || backTitle, state),
-      headerRight: getValue(navigationParams.right || right || rightButton || params.renderRightButton, state),
+      headerBackTitleEnabled: navigationParams.backTitleEnabled || backTitleEnabled,
+      headerLayoutPreset: navigationParams.headerLayoutPreset || headerLayoutPreset,
       headerLeft: getValue(navigationParams.left || left || leftButton || params.renderLeftButton, state),
-      headerTitle: getValue(navigationParams.renderTitle || renderTitle || params.renderTitle, state),
```
Comment (aksonov) [inline src/navigationStore.js]: Any reason to remove headerTitle?

Comment (daviscabral) [reply src/navigationStore.js]: It was not removed - just ordered those items alphabetically.

File: src/navigationStore.js [hunk 2]
```diff
@@ -195,15 +197,17 @@ function createNavigationOptions(params) {
...
+      headerRight: getValue(navigationParams.right || right || rightButton || params.renderRightButton, state),
       headerStyle: getValue(navigationParams.headerStyle || headerStyle || navigationBarStyle, state),
-      headerBackImage: navigationParams.backButtonImage || backButtonImage,
+      headerTintColor: navBarButtonColor || props.tintColor || navigationParams.tintColor || navigationParams.headerTintColor,
+      headerTitle: getValue(navigationParams.renderTitle || renderTitle || params.renderTitle, state),
```
Comment (daviscabral) [inline src/navigationStore.js]: @aksonov ^

**Pull request merged**